File: backend/models/advice.py
from pydantic import BaseModel

class TripAdviceRequest(BaseModel):
    activity: str
    date: str
    weather: dict
    score: dict

class TripAdvice(BaseModel):
    summary: str
    gear: list[str]
    expectations: list[str]
    cautions: list[str]

File: frontend/src/types/advice.ts
import type { Activity, DailyForecast, Score } from "./forecast"

export type TripAdviceRequest = {
    activity: Activity
    date: string
    weather: DailyForecast
    score: Score
}

export type TripAdvice = {
    summary: string
    gear: string[]
    expectations: string[]
    cautions: string[]
}